compaction/time_window_compaction_strategy.cc

```diff
@@ -220,14 +220,11 @@ time_window_compaction_strategy::get_sstables_for_compaction(column_family& cf,
     }
 
     if (!expired.empty()) {
-        auto is_expired = [&] (const shared_sstable& s) { return expired.contains(s); };
-        candidates.erase(boost::remove_if(candidates, is_expired), candidates.end());
+        clogger.debug("Going to compact {} expired sstables", expired.size());
+        return compaction_descriptor(compaction_descriptor::compact_fully_expired::yes, std::vector<shared_sstable>(expired.begin(), expired.end()), cf.get_sstable_set(), service::get_local_compaction_priority());
     }
 
     auto compaction_candidates = get_next_non_expired_sstables(cf, std::move(candidates), gc_before);
-    if (!expired.empty()) {
-        compaction_candidates.insert(compaction_candidates.end(), expired.begin(), expired.end());
-    }
```

Comment:
direction of patch looks good me.

only had a couple of minor comments.

makes sense to serialize fully expired jobs with weight 0 as they'll run very quickly and shouldn't be blocked on other weights.

by segregating fully expired sstables into a separate job, we now will release them quicker and by using weight 0 we'll guarantee that they'll be submitted for compaction even when under heavy load. so we won't have to retry them at another interval which is controlled by strategy option.